tests/nested/core/core20-basic/task.yaml

```diff
@@ -44,7 +44,7 @@ execute: |
     fi
 
     echo "Wait for device initialisation to be done"
-    nested_retry_until_success 10 5 "snap changes | grep -q 'Done.*Initialize device'"
+    nested_retry "--wait 5 -n 10 sh -c 'snap changes | grep -qE \"Done.*Initialize device\"'"
```

Comment:
This is getting out of hand with the quoting. Instead of trying to bend the shell to do something reasonable, why not make retry ssh aware? We could tweak the retry script to take for instance
```
--ssh-host <host:port> --ssh-ident user1:ubuntu --wait 5 -n 10 sh -c ...
```

The nested.sh file could provide NESTED_HOST=localhost:$NESTED_PORT, NESTED_USER=user1:ubuntu. The retry script then then use eg. paramiko library to run the command. The library should be available in all systems we use.
The commands could then become
```
retry --ssh-host $NESTED_HOST --ssh-ident=$NESTED_IDENT \
  --wait 5 -n 10 \
  sh -c 'snap changes | grep -qE "Done.*Initialize device"'
```
No crazy quoting.

If you need to pass environment variables, we could add `--export <ARG>`, eg. `--export REFRESH_ID` which would pass the command to the environment of the remote process.

Reply:
@bboozzoo finally I implemented that in a different way to avoid making more complex the retry and reusing existent nested_exec and retry tool, could you please take a quick look and give your opinion?